sdk/instance/backend.go

```diff
@@ -46,7 +48,12 @@ func (s *Backend) querier(request cosmostypes.Request, path []string, req abci.R
 		}
 		return s.Get(request, hash)
 	case "list":
-		return s.List(request)
+		var f api.ListInstanceRequest_Filter
+		if err := json.Unmarshal(req.Data, &f); err != nil {
```

Comment:
could you move the unmarshal in:
https://github.com/mesg-foundation/engine/blob/940abdaa67779f39482e03b0bb6afbac0a05b745/cosmos/module.go#L112-L127

this function already marshal the data before sending it back